adafruit_rfm69.py

```diff
@@ -751,11 +778,17 @@ def send(
         self.transmit()
         # Wait for packet sent interrupt with explicit polling (not ideal but
         # best that can be done right now without interrupts).
-        start = time.monotonic()
         timed_out = False
-        while not timed_out and not self.packet_sent():
-            if (time.monotonic() - start) >= self.xmit_timeout:
-                timed_out = True
+        if HAS_SUPERVISOR:
+            start = supervisor.ticks_ms()
+            while not timed_out and not self.packet_sent():
+                if ticks_diff(supervisor.ticks_ms(), start) >= self.xmit_timeout * 1000:
+                    timed_out = True
+        else:
+            start = time.monotonic()
+            while not timed_out and not self.packet_sent():
+                if time.monotonic() - start >= self.xmit_timeout:
+                    timed_out = True
```

Comment:
Want to factor this out with the chunk below? You should pass in `self.packet_sent` (no ()) and call it in the factored out function.

Reply:
Good idea -- I work on this.
Note, once I get this cleaned up, I'll go back ad make the same changes in the RFM9x. Sorry I did not wait longer before merging it.